src/tests/test_run.py

```diff
@@ -88,9 +89,11 @@ def test_is_postgres_or_redshift(dialect):
 
 
 def test_handle_postgres_special(mock_conns):
-    with pytest.raises(ImportError):
+    with pytest.raises(UsageError) as excinfo:
```

Comment:
We are capturing Ipython's UsageError which is raised by the factory but this way we might miss the actual error we wanted to raise.

e.g. `test_handle_postgres_special` raises `exceptions.MissingPackageError("pgspecial not installed")` but the test will also be positive for `exceptions.ArgumentError("pgspecial not installed")`.

If our goal is to eventually replace all errors in our system with those from this factory, we will need to differentiate between them based on their inner messages. Since these messages may change more frequently than the error name (which is the type) - do you think we can test by the exception name? maybe create a custom error context to check the error by name as well.

Reply:
fair point. this testing is a bit awkward since we are always testing against UsageError due to an IPython limitation. I modified the existing tests so they also check the `.error_type` attribute.